### PR TITLE
Improve the i18n messages creation.

### PR DESCRIPTION
**Pull request by JimiC**

### CODE REVIEW

**Comment by JimiC** [comment]
Travis has some issues atm.

File: src/i18n/langResources/lang.de.ts [hunk 1]
```diff
@@ -20,19 +20,22 @@ export const langDe: ILangResource = {
...
-  tsOfficialPresetMessage: 'Offizielles TS Icon wird',
-  jsonOfficialPresetMessage: 'Offizielles JSON Icon wird',
-  hideFoldersPresetMessage: 'Ordner Icons werden',
-  enabled: 'aktiviert',
-  disabled: 'deaktiviert',
+  iconCustomization: 'Das Icon-Manifest wird regeneriert. ',
+  iconRestore: 'Das Icon-Manifest wird zurückgesetzt. ',
+  ngPresetEnabled: 'Angular Icons wird aktiviert. ',
+  ngPresetDisabled: 'Angular Icons wird deaktiviert. ',
+  jsOfficialPresetEnabled: 'Offizielles JS Icon wird aktiviert. ',
```
Comment (jens1o) [inline src/i18n/langResources/lang.de.ts]: Thanks! But is the space at the end really needed?

File: src/i18n/langResources/lang.en.ts [hunk 2]
```diff
@@ -20,19 +20,22 @@ export const langEn: ILangResource = {
...
-  jsOfficialPresetMessage: 'Official JS icon will be',
-  tsOfficialPresetMessage: 'Official TS icon will be',
-  jsonOfficialPresetMessage: 'Official JSON icon will be',
-  hideFoldersPresetMessage: 'Folder icons visibility will be',
-  enabled: 'enabled',
-  disabled: 'disabled',
+  iconCustomization: 'The icons manifest will be regenerated.',
+  iconRestore: 'The icons manifest will be restored.',
+  ngPresetEnabled: 'Angular icons will be enabled.',
+  ngPresetDisabled: 'Angular icons will be disabled.',
```
Comment (jens1o) [inline src/i18n/langResources/lang.en.ts]: there are no spaces at the end like in German.

File: src/i18n/langResources/lang.es.ts [hunk 3]
```diff
@@ -20,19 +20,22 @@ export const langEs: ILangResource = {
...
-  iconRestoreMessage: 'El manifiesto de iconos se va a restaurar. ',
-  ngPresetMessage: 'Los iconos de Angular se van a',
-  jsOfficialPresetMessage: 'El icono oficial de JS se va a',
-  tsOfficialPresetMessage: 'El icono oficial de TS se va a',
-  jsonOfficialPresetMessage: 'El icono oficial de JSON se va a',
-  hideFoldersPresetMessage: 'La visibilidad de los iconos de Carpeta se va a',
-  enabled: 'activar',
-  disabled: 'desactivar',
+  iconCustomization: 'El manifiesto de iconos se va a regenerar.',
+  iconRestore: 'El manifiesto de iconos se va a restaurar.',
```
Comment (jens1o) [inline src/i18n/langResources/lang.es.ts]: no spaces

File: src/i18n/langResources/lang.fr.ts [hunk 4]
```diff
@@ -20,19 +20,22 @@ export const langFr: ILangResource = {
...
-  enabled: 'activée(s)',
-  disabled: 'désactivée(s)',
+  iconCustomization: 'Le manifeste des icônes sera régénéré.',
+  iconRestore: 'Le manifeste des icônes sera restauré.',
+  ngPresetEnabled: 'Les icônes Angular seront activées.',
+  ngPresetDisabled: 'Les icônes Angular seront désactivées.',
+  jsOfficialPresetEnabled: 'L\'icône officielle JS sera activée.',
+  jsOfficialPresetDisabled: 'L\'icône officielle JS sera désactivée.',
+  tsOfficialPresetEnabled: 'L\'icône officielle TS sera activée.',
+  tsOfficialPresetDisabled: 'L\'icône officielle TS sera désactivée.',
```
Comment (jens1o) [inline src/i18n/langResources/lang.fr.ts]: eh, no spaces?

File: src/i18n/langResources/lang.de.ts (hunk 1 above)
Comment (JimiC) [inline src/i18n/langResources/lang.de.ts]: Good catch. Removed.

File: src/i18n/langResources/lang.en.ts (hunk 2 above)
Comment (JimiC) [inline src/i18n/langResources/lang.en.ts]: That's why I named it `improvement.`

File: src/i18n/langResources/lang.fr.ts (hunk 4 above)
Comment (JimiC) [inline src/i18n/langResources/lang.fr.ts]: They are added internally.

File: src/i18n/langResources/lang.de.ts [hunk 5]
```diff
@@ -20,18 +20,18 @@ export const langDe: ILangResource = {
...
-  jsOfficialPresetDisabled: 'Offizielles JS Icon wird deaktiviert. ',
-  tsOfficialPresetEnabled: 'Offizielles TS Icon wird aktiviert. ',
-  tsOfficialPresetDisabled: 'Offizielles TS Icon wird deaktiviert. ',
-  jsonOfficialPresetEnabled: 'Offizielles JSON Icon wird aktiviert. ',
-  jsonOfficialPresetDisabled: 'Offizielles JSON Icon wird deaktiviert. ',
-  hideFoldersPresetEnabled: 'Ordner Icons werden aktiviert. ',
-  hideFoldersPresetDisabled: 'Ordner Icons werden deaktiviert. ',
+  iconCustomization: 'Das Icon-Manifest wird regeneriert.',
+  iconRestore: 'Das Icon-Manifest wird zurückgesetzt.',
+  ngPresetEnabled: 'Angular Icons wird aktiviert ',
```
Comment (jens1o) [inline src/i18n/langResources/lang.de.ts]: Missing dot.

File: src/i18n/langResources/lang.de.ts [hunk 6]
```diff
@@ -20,18 +20,18 @@ export const langDe: ILangResource = {
...
+  iconRestore: 'Das Icon-Manifest wird zurückgesetzt.',
+  ngPresetEnabled: 'Angular Icons wird aktiviert.',
+  ngPresetDisabled: 'Angular Icons wird deaktiviert.',
+  jsOfficialPresetEnabled: 'Offizielles JS Icon wird aktiviert.',
+  jsOfficialPresetDisabled: 'Offizielles JS Icon wird deaktiviert.',
+  tsOfficialPresetEnabled: 'Offizielles TS Icon wird aktiviert.',
+  tsOfficialPresetDisabled: 'Offizielles TS Icon wird deaktiviert.',
+  jsonOfficialPresetEnabled: 'Offizielles JSON Icon wird aktiviert.',
+  jsonOfficialPresetDisabled: 'Offizielles JSON Icon wird deaktiviert.',
+  hideFoldersPresetEnabled: 'Ordner Icons werden aktiviert.',
```
Comment (jens1o) [inline src/i18n/langResources/lang.de.ts]: wut? Wrong phrase here.

Comment (JimiC) [inline src/i18n/langResources/lang.de.ts]: And that's what the toggle does.

Comment (JimiC) [inline src/i18n/langResources/lang.de.ts]: Oh you need it should be reverted.

File: src/commands/index.ts [hunk 7]
```diff
@@ -65,25 +65,18 @@ function resetProjectDetectionDefaultsCommand(): void {
 
 function togglePreset(
   preset: string,
-  presetMessage: string,
+  presetKey: string,
   reverseAction: boolean = false,
   global: boolean = true): void {
 
   const value = getToggleValue(preset);
+  const action = reverseAction ? value ? 'Disabled' : 'Enabled' : value ? 'Enabled' : 'Disabled';
```
Comment (jens1o) [inline src/commands/index.ts]: could you add parentheses please to improve readability?

Comment (JimiC) [inline src/commands/index.ts]: No parentheses needed. https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

Comment (JimiC) [inline src/commands/index.ts]: What I could do though is change the formatting to:
```
const action = reverseAction
    ? value 
        ? 'Disabled'
        : 'Enabled'
    : value 
         ? 'Enabled'
         : 'Disabled';
```

Comment (jens1o) [inline src/commands/index.ts]: I just want to improve the readability, I know that these are optional. But first I was confused to read it a little bit.

I would like it like this:
```js
const action = reverseAction ? (value ? 'Disabled' : 'Enabled') : (value ? 'Enabled' : 'Disabled');
```

But I'm also fine with your idea.

File: src/i18n/langResources/lang.de.ts (hunk 6 above)
Comment (robertohuertasm) [inline src/i18n/langResources/lang.de.ts]: Are we good with this translation?

Comment (jens1o) [inline src/i18n/langResources/lang.de.ts]: yes, we are. I'm submitting the German translation in the day, but I have to learn for school.

Comment (JimiC) [inline src/i18n/langResources/lang.de.ts]: @robertohuertasm Obviously Jens when saying 
>I'm submitting the German translation in the day, but I have to learn for school.

is referring to the translation for the `package.nls.json`.

**Pull request merged**